src/Microsoft.AspNetCore.Razor.Design/build/netstandard2.0/Microsoft.AspNetCore.Razor.Design.Compilation.targets

```diff
@@ -1,76 +1,114 @@
-
-    The XamlPreCompile target must remain identical to
-    the CoreCompile target in Microsoft.CSharp.Core.targets.
-    Any updates to one must be made to the other.
-  -->
-
-
   <ItemGroup>
     <!-- Used to creating the final compiled Razor dll -->
-    <_RazorIntermediateAssembly Include="$(IntermediateOutputPath)$(RazorTargetName).dll" />
+    <RazorIntermediateAssembly Include="$(IntermediateOutputPath)$(RazorTargetName).dll" />
 
     <!-- Similar to https://github.com/Microsoft/msbuild/blob/908cc9ccd4961441628f68e37a148183a87bb067/src/Tasks/Microsoft.Common.CurrentVersion.targets#L146-L153 -->
-    <_RazorIntermediatePdb 
+    <_RazorDebugSymbolsIntermediatePath 
       Condition="'$(DebugSymbols)'=='true' and '$(DebugType)'!='' and '$(DebugType)'!='none' and '$(DebugType)'!='embedded'" 
       Include="$(IntermediateOutputPath)$(RazorTargetName).pdb" />
   </ItemGroup>
- 
-  <Target
-    Name="RazorCoreCompile"
-    Inputs="$(MSBuildAllProjects);
-            @(RazorCompile);
-            @(_CoreCompileResourceInputs);
-            $(ApplicationIcon);
-            $(AssemblyOriginatorKeyFile);
-            @(RazorReferencePath);"
-              
-    Outputs="@(_RazorIntermediateAssembly);
-             @(_RazorIntermediatePdb)"
-    Returns=""
-    DependsOnTargets="_RazorResolveTagHelpers;RazorCoreGenerate">
 
+  <!--
+  What follows in this file is based on:
+     https://github.com/dotnet/roslyn/blob/4d92b18aee99ba8b1b4770ce65133e9ca65a94fe/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets 
+
+  We need to keep this basically up to date, as well as track the set of modifications we've made. Try to keep the formatting
+  similar to the original to reduce noise. In general try to only deviate from the CoreCompile target when we need to for
+  correctness.
+
+  We also want to avoid doubling up on things that don't make a ton of sense in the Razor views assembly, like
+  embedded files and resources, these are already present in the application's assembly.
+
+  Changes:
+    Name="RazorCoreCompile" and DependsOnTarget="RazorCoreGenerate"
+    
+    Replace @(Compile) with @(RazorCompile)
+    Replace @(_DebugSymbolsIntermediatePath) with @(_RazorDebugSymbolsIntermediatePath)
+    Replace @(IntermediateAssembly) with @(RazorIntermediateAssembly)
+    Replace @(ReferencePathWithRefAssemblies) with @(RazorReferencePath)
+
+    Set TargetType="$(OutputType)" to TargetType="Library" - Razor is always a .dll
+
+    Remove Returns="@(CscCommandLineArgs)"
+    Remove @(_CoreCompileResourceInputs)
+    Remove $(ApplicationIcon) $(Win32Resource) $(Win32Manifest)
+    Remove @(EmbeddedDocumentation) and @(EmbeddedFiles)
+    Remove @(CustomAdditionalCompileInputs) and @(CustomAdditionalCompileOutputs)
+    Remove @(DocFileItem)
+    Remove PdbFile="$(PdbFile)"
+    Remove OutputRefAssembly="@(IntermediateRefAssembly)"
+
+    Remove EmbedAllSources="$(EmbedAllSources)" - not supported by our supported version of MSBuild
```

Comment:
This is a concern with us shipping outside of the SDK... We have to think kinda carefully about what versions of the SDK we can support. I don't want to block on it right now - but we need to know the answer before we ship 2.1.

My expectation is that we'll end up with these files split between the SDK and a nupkg.